### PR TITLE
Add NodeJS install resource

### PR DESCRIPTION
**Pull request by elliott-davis**
@chef-cookbooks/delivery-migration-team

### CODE REVIEW

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

File: test/fixtures/cookbooks/languages_node/recipes/default.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+include_recipe 'chef-sugar::default'
+include_recipe 'languages::default'
+
+node_install 'v4.1.2'
```
Comment (tylercloke) [inline test/fixtures/cookbooks/languages_node/recipes/default.rb]: I would add a test for a non-default prefix.

Comment (tylercloke) [inline test/fixtures/cookbooks/languages_node/recipes/default.rb]: Just install two versions of node (default and non-default), and then test for both existing in the file below.

File: libraries/node_install.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        source 'https://raw.githubusercontent.com/creationix/nvm/v0.29.0/install.sh'
+        mode '0755'
+        not_if { ::File.exist?("#{prefix}/nvm") }
+      end
+      execute 'nvm-install' do
+        command "NVM_DIR=#{prefix}/nvm /tmp/nvm_install.sh"
+        not_if { ::File.exist?("#{prefix}/nvm") }
+      end
+      bash 'install-node' do
+        code <<-CODE.gsub(/^ {10}/, '')
```
Comment (tylercloke) [inline libraries/node_install.rb]: Might wanna throw a comment in here about what this gsub is doing.

**Comment by tylercloke** [comment]
Generally LGTM other than a few comments around testing, etc. Looks good for linux!

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

**Comment by ksubrama** [comment]
:+1:

**Comment by schisamo** [comment]
Chatted with @elliott-davis about a few refinements we might want to make at a future date but let's move ahead with this for now. :+1:

**Comment by schisamo** [comment]
@delivery approve

**Comment by chef-delivery** [comment]
Change: ac606cb5-9933-45be-a0ed-331c5b17264b approved by: @schisamo

**Comment by chef-delivery** [comment]
This PR has passed 'Acceptance' and is ready to be delivered!
  Use: '@delivery deliver' when validated in acceptance.

**Comment by schisamo** [comment]
@delivery deliver

**Comment by chef-delivery** [comment]
Change: ac606cb5-9933-45be-a0ed-331c5b17264b delivered by: @schisamo

**Pull request merged**